lib/pipeline.ts

```diff
@@ -162,12 +160,15 @@ export interface PipelineProps {
    * @default - Duration.hours(8)
    */
   readonly buildTimeout?: Duration;
+}
 
-  /**
-   * Slack channel configurations where failure notification
-   * in this pipeline should be sent.
-   */
-  readonly failureNotifySlack?: chatbot.SlackChannelConfiguration[];
+export interface PipelineNotificationBindOptions {
+  readonly pipeline: Pipeline;
+  readonly codePipeline: cpipeline.Pipeline;
+}
+
+export interface IPipelineNotification {
+  bind(pipeline: PipelineNotificationBindOptions): void;
```

Comment:
Traditionally, `bind()` takes a `scope: IConstruct`, which is where it will create constructs.

Yes we'll call it with `bind(this, this)` but that still feels more correct and allows for more flexibility.